apps/src/tutorialExplorer/tutorialSet.jsx

```diff
@@ -13,18 +24,61 @@ const TutorialSet = React.createClass({
     disabledTutorials: React.PropTypes.arrayOf(React.PropTypes.string).isRequired
   },
 
+  getInitialState() {
+    return {
+      showingDetail: false,
+      chosenItem: null
+    };
+  },
+
+  tutorialClicked(item) {
+    this.setState({showingDetail: true, chosenItem: item});
+  },
+
+  tutorialDetailClosed() {
+    this.setState({showingDetail: false, chosenItem: null});
+  },
+
+  changeTutorial(direction) {
+    const index = this.props.tutorials.indexOf(this.state.chosenItem);
+    let nextItem = null;
+    if (direction === "next" && index < this.props.tutorials.length - 1) {
+      nextItem = this.props.tutorials[index + 1];
+    } else if (direction === "previous" && index > 0) {
+      nextItem = this.props.tutorials[index - 1];
+    }
+    if (nextItem) {
+      this.setState({showingDetail: true, chosenItem: nextItem});
+    }
+  },
+
   render() {
+    const disabledTutorial = this.state.showingDetail &&
+      this.props.disabledTutorials.indexOf(this.state.chosenItem.short_code) !== -1;
+
     return (
-      <div style={{overflow: "hidden"}}>
-        {this.props.tutorials.map(item => (
+      <div>
+        <TutorialDetail
+          showing={this.state.showingDetail}
+          item={this.state.chosenItem}
+          closeClicked={this.tutorialDetailClosed}
+          localeEnglish={this.props.localeEnglish}
+          disabledTutorial={disabledTutorial}
+          changeTutorial={this.changeTutorial}
+        />
+        {this.props.tutorials.length > 0 && this.props.tutorials.map(item => (
```

Comment:
I would think that the .length check is unnecessary, as [].map should give you []. (Unless maybe React tries to render [] as something and but doesnt with false?)